Sources/Core/Sources/ECS/Systems/PlayerInputSystem.swift

```diff
@@ -146,6 +148,47 @@ public final class PlayerInputSystem: System {
         if !message.isEmpty {
           guiState.messageInput?.removeLast()
         }
+      } else if event.key == .upArrow {
+        // If no message is selected, select the above message
+        if guiState.currentMessageIndex == nil {
+          guiState.stashedMessageInput = guiState.messageInput
+          // If there are no messages, do nothing
+          if guiState.playerMessageHistory.count > 0 {
+            // Index up a message
+            guiState.currentMessageIndex = guiState.playerMessageHistory.count - 1
+            if guiState.currentMessageIndex == guiState.currentMessageIndex {
+              // Set the message input to the message at the index
+              guiState.messageInput = guiState.playerMessageHistory[guiState.currentMessageIndex!]
+            }
+          }
+        } else {
+          // If there is a message selected, index up a message
+          if guiState.currentMessageIndex! > 0 {
+            guiState.currentMessageIndex! -= 1
+          }
+          if guiState.currentMessageIndex == guiState.currentMessageIndex {
```

Comment:
Again over here. Apologies if it’s my mistake but it doesn’t really look right

Reply:
same as earlier